Signal/src/ViewControllers/AppSettingsViewController.m

```diff
@@ -238,6 +211,90 @@ - (void)updateTableContents
+
+    const NSUInteger kAvatarSize = 68;
+    // TODO: Replace this icon.
+    UIImage *_Nullable localProfileAvatarImage = [OWSProfileManager.sharedManager localProfileAvatarImage];
+    UIImage *avatarImage = (localProfileAvatarImage
+            ?: [[UIImage imageNamed:@"profile_avatar_default"]
+                   imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]);
+    OWSAssert(avatarImage);
+
+    AvatarImageView *avatarView = [[AvatarImageView alloc] initWithImage:avatarImage];
+    if (!localProfileAvatarImage) {
+        avatarView.tintColor = [UIColor colorWithRGBHex:0x888888];
+    }
+    [cell.contentView addSubview:avatarView];
+    [avatarView autoVCenterInSuperview];
+    [avatarView autoPinLeadingToSuperView];
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+
+    if (!localProfileAvatarImage) {
+        UIImage *cameraImage = [UIImage imageNamed:@"settings-avatar-camera"];
+        cameraImage = [cameraImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+        UIImageView *cameraImageView = [[UIImageView alloc] initWithImage:cameraImage];
+        cameraImageView.tintColor = [UIColor ows_materialBlueColor];
+        [cell.contentView addSubview:cameraImageView];
+        [cameraImageView autoPinTrailingToView:avatarView];
+        [cameraImageView autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:avatarView];
+    }
+
+    UIView *threadNameView = [UIView containerView];
+    [cell.contentView addSubview:threadNameView];
+    [threadNameView autoVCenterInSuperview];
+    [threadNameView autoPinTrailingToSuperView];
+    [threadNameView autoPinLeadingToTrailingOfView:avatarView margin:16.f];
+
+    UILabel *threadTitleLabel = [UILabel new];
+    NSString *_Nullable localProfileName = [OWSProfileManager.sharedManager localProfileName];
+    if (localProfileName.length > 0) {
+        threadTitleLabel.text = localProfileName;
+        threadTitleLabel.textColor = [UIColor blackColor];
+        threadTitleLabel.font = [UIFont ows_dynamicTypeTitle2Font];
+    } else {
+        threadTitleLabel.text = NSLocalizedString(
+            @"APP_SETTINGS_EDIT_PROFILE_NAME_PROMPT", @"Text prompting user to edit their profile name.");
+        threadTitleLabel.textColor = [UIColor ows_materialBlueColor];
+        threadTitleLabel.font = [UIFont ows_dynamicTypeTitle3Font];
+        DDLogError(@"threadTitleLabel.font: %f", threadTitleLabel.font.pointSize);
+    }
+    threadTitleLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    [threadNameView addSubview:threadTitleLabel];
+    [threadTitleLabel autoPinEdgeToSuperviewEdge:ALEdgeTop];
+    [threadTitleLabel autoPinWidthToSuperview];
+
+    __block UIView *lastTitleView = threadTitleLabel;
+
+    const CGFloat kSubtitlePointSize = 12.f;
+    void (^addSubtitle)(NSAttributedString *) = ^(NSAttributedString *subtitle) {
```

Comment:
I'm unclear on the purpose of this block. Can you comment about why it's a block vs just inlining it?